core/src/main/java/org/infinispan/metrics/impl/MetricsCollector.java

```diff
@@ -181,8 +181,8 @@ private static String generateRandomName() {
       return metricIds;
    }
 
-   public void unregisterMetric(MetricID metricId) {
-      boolean removed = registry.remove(metricId);
+   public void unregisterMetric(Object metricId) {
```

Comment:
Is this change really necessary? I saw that it seems to be using MetricID everywhere and if not this change is setting ourselves up for a ClassCastException with no compiler checks :(

Reply:
MetricID is a smallrye-metrics implementation detail which should not leak everywhere. Since we will be migrating to micrometer, we can handle this as an opaque type

Reply:
Okay, I just worry because the line below does a cast, so we lose that type info but still rely upon the concrete class. I guess you are doing this for an upcoming change though.

Reply:
Yeah, the task is https://issues.redhat.com/browse/ISPN-13420